### DISCUSSION TITLE
Logging, canonical log lines and seeing all lines for a single request

### DISCUSSION THREAD

**Discussion by sandstrom** [grafana/grafana#39129] [General]
We usually need to see all log lines for a particular request, and we give all log lines from a particular request the same unique request id (trace id).

What we're missing is some UI around this use-case in Grafana. For example, if we could click any label  (or calculated label) among Loki log lines and quickly bring up all the logs with that filter value that are before and after the current line (e.g. +/- 10 lines, and a "more" button with which one could expand further), that would be very useful.

Similar to the [context viewer](https://grafana.com/blog/2019/08/14/new-in-grafana-v6.3-introducing-lokis-log-row-context-viewer/), but more intelligent. We almost always investigate logs in terms of a "request" or "async job", not each line by themselves.

This would make it easy to e.g. "expand" a request id and see all log lines for that request id. Makes sense if you are using e.g. [Canonical Log Lines](https://brandur.org/canonical-log-lines).

Moved here: https://github.com/grafana/grafana/issues/82515

**Comment by 09jvilla** [comment]
@sandstrom Can you help me understand the use case a bit more? 

If you click the magnifying glass with a plus sign in the UI, that will modify your query so you only search for log streams with that label. 

In my screenshot below, if I hit the magnifying glass with the plus sign next to "probe" I'd only see log streams where "probe=Mumbai." 

<img width="538" alt="Screen Shot 2021-11-22 at 2 00 46 PM" src="https://user-images.githubusercontent.com/9610816/142927139-54fa1977-34ee-4f90-b122-d8452de8734d.png">

Could you not do that with your needs? And/or could you not just add `|= "traceid_value"` to your logql query (where "traceid_value" is the request id you're looking at)?

**Reply by sandstrom** [reply]
Yes, with the "magnifying glass" now working on dynamic/computed labels too, that solves our problem.

**Reply by 09jvilla** [reply]
So are you saying we can close this issue? Or is there something you still think is missing?

**Reply by sandstrom** [reply]
Yes, you can close

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
